src/clients/Deluge.ts

```diff
@@ -233,32 +247,53 @@ export default class Deluge implements TorrentClient {
 	/**
 	 * if Label plugin is loaded, adds (if necessary)
 	 * and sets the label based on torrent hash.
+	 * @param newTorrent the searchee of the newTorrent
+	 * @param label the destination label for the newTorrent/searchee
 	 */
-	private async setLabel(
-		name: string,
-		infoHash: string,
-		label: string,
-	): Promise<void> {
+	private async setLabel(newTorrent: Searchee, label: string): Promise<void> {
+		let setResult: Result<void, ErrorType>;
 		if (!this.isLabelEnabled) return;
 		try {
-			const setResult = await this.call<void>("label.set_torrent", [
-				infoHash,
-				label,
-			]);
-			if (setResult.error?.code === DelugeErrorCode.RPC_FAIL) {
+			const getCurrentLabels = await this.call<string[]>(
+				"label.get_labels",
+				[],
+			);
+			if (getCurrentLabels.isErr()) {
+				this.isLabelEnabled = false;
+				throw new Error("Labels have been disabled.");
```

Comment:
can we just silently fail if labeling is not enabled? I don't think it needs to be required in order to use cross-seed

Reply:
This was added in the case that the label plugin was disabled or failed in an irrecoverable manner (for some reason) after cross-seed had started and determined that it was loaded.